src/content_script/mastodonInject.js

```diff
@@ -121,28 +118,47 @@ async function injectInteractionButtons() {
     });
 }
 
+/**
+ * Initialise injection for feedElement.
+ * 
+ * @returns {void}
+ */
+async function injectFeed() {
+    const observer = new MutationObserver(() => {
+        injectInteractionButtons();
+    });
+
+    try {
+        const feedElement = await waitForElement(
+            "#mastodon .item-list[role='feed']",
+            false,
+        );
+
+        observer.observe(feedElement, {
+            childList: true, subtree: true,
+        });
+    } catch (error){
+        // feedElement not found
+    }
+}
+
 /**
  * Initialise injection for all remote Mastodon buttons.
  * 
  * @returns {void}
  */
-async function init() {
+function init() {
```

Comment:
Same here, let's not ignore the promises like that, AFAIK a simple `init().catch(console.error).then(console.log)` (the log being rather optional) could also help to catch things and also properly log them. (In the last line.)

That should also ignore the Promise.